src/Parser/Mage/Fire/Modules/Features/Cinderstorm.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import Wrapper from 'common/Wrapper';
+import { formatNumber, formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Analyzer from 'Parser/Core/Analyzer';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+
+const DAMAGE_WINDOW_MS = 5000;
+
+class Cinderstorm extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    abilityTracker: AbilityTracker,
+  };
+
+  cinderHits = 0;
+  totalHits = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.CINDERSTORM_TALENT.id);
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.CINDERSTORM_TALENT.id) {
+      return;
+    }
+    this.castTimestamp = event.timestamp;
+    this.cinderHits = 0;
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.CINDERSTORM_DAMAGE.id) {
+      return;
+    }
+    if (this.castTimestamp - event.timestamp < DAMAGE_WINDOW_MS && this.cinderHits < 6) {
+      this.cinderHits += 1;
+      this.totalHits += 1;
+    }
+  }
+
+  get cindersHitPercent() {
+    return this.totalHits / this.totalCinders;
+  }
+
+  get totalCinders() {
+    return this.abilityTracker.getAbility(SPELLS.CINDERSTORM_TALENT.id).casts * 6;
+  }
+
+  get missedCinders() {
+    return this.totalCinders - this.totalHits;
+  }
+
+  get missedCindersPercent() {
+    return this.missedCinders / this.totalCinders;
+  }
+
+  get averageHitsPerCast() {
+    return 6 - ((this.totalHits / 6) / (this.totalCinders / 6));
```

Comment:
I don't understand how this could be the correct calculation.

Lets take a simple example: you cast Cinderstorm once and completely miss. This calculation is then `6 - ((0 / 6) - (6 / 6)) === 6 - (0 / 6) === 6` ... that seems obviously incorrect. Wouldn't the right calculation be much simpler, something like `this.totalHits / this.abilityTracker.getAbility(SPELLS.CINDERSTORM_TALENT.id).casts` ?

Reply:
yea i was overthinking it and still got it wrong. ill fix it.